src/yadmark/benchmark/external/apfelpy_utils.py

```diff
@@ -334,8 +385,13 @@ def compute_apfelpy_data(theory, observables, pdf):
                 )
                 tab_sf = tabulate_nc(ap, obs_name, sfobj, NQ, QMin, QMax, thrs)
 
+            # shift convolution for massive
+            eta = 1.0
+            if fns == "FFNS":
+                m_h = masses[3] if heaviness == "charm" else masses[4]
+                eta = Q2 / (Q2 + 4 * m_h**2)
             # compute the actual result
-            result = tab_sf.EvaluatexQ(x, np.sqrt(Q2))
+            result = tab_sf.EvaluatexQ(x / eta, np.sqrt(Q2))
```

Comment:
okay I copied the syntax, from apfel++ ...

Reply:
your fix [781ed78](https://github.com/NNPDF/yadism/pull/245/commits/781ed78e1926c9d1592aa07e3831788351d526e5) actually only renames eta :upside_down_face: but I wanted to shift the division inside the `if` (sorry github would not let me suggest the right code, because he thinks there were deleted lines - I'm wondering if I should file a bug report against github ...)

Reply:
I see sorry.